### DISCUSSION TITLE
React azure static web app successful build but blank page

### DISCUSSION THREAD

**Discussion by realpinman** [community/community#72669] [Actions]
### Select Topic Area

Question

### Body

I have a react vite visual studio project with a CI/CD pipeline that publishes on successful commit however when i view my app i get a blank page (although static content like text is displayed fine).  It's as though the project does not build even though i have included a app_build_command. my action file is below. 

`  push:
    branches:
      - master
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - master

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: true
          lfs: false
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.***** }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match your app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/EliteAccountReact" # App source code path
          api_location: "" # Api source code path - optional
          output_location: "/dist" # Built app content directory - optional
          app_build_command: "npm run build" # Build command - optional
          ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.**** }}
          action: "close"`

Also when i view my page i get an error in the console "Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec." so it seems the server cannot serve javascrpipt maybe. Any help would be appreciated.

**Comment by oussamaef06** [comment]
The error message you're seeing, "Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of 'application/octet-stream'," suggests an issue with the MIME type configuration when serving JavaScript files. This error can lead to a blank page in your application. To resolve this issue, you can take the following steps:

Check the Content-Type Header:

Ensure that your server is sending the correct Content-Type header when serving JavaScript files. It should be set to "application/javascript" for JavaScript files.
You can configure this in your web server or hosting environment. If you are using a CDN, make sure the CDN settings are correct.
Review Your Build Configuration:

Double-check your build configuration in your Vite or React project. Make sure it is generating JavaScript files with the .js extension.
If your project is set up to use ES modules, the files should be served as module scripts, which are typically identified by the .mjs extension.
Check for Server-Side MIME Type Configuration:

Verify the server or hosting environment where your React application is deployed. Some server configurations can impact the MIME types of served files.
Look for any custom configurations or server settings that might be altering the MIME types of files.
Ensure Correct Deployment Paths:

Double-check that your paths and file locations are correct in your deployment configuration. The app_location, output_location, and app_build_command values in your GitHub Actions file should match the actual structure of your project.
CORS and Security Headers: Ensure that your server or CDN is configured to allow the correct Cross-Origin Resource Sharing (CORS) headers and security headers like Content-Security-Policy. Misconfigured security headers can sometimes interfere with loading JavaScript files.

Bundle Configuration: If you're using Vite, check your Vite configuration to ensure it generates JavaScript files as expected and that any bundling settings are correct.

Debugging: Use the browser's developer tools to inspect the network requests and responses for JavaScript files. This can help you identify if the issue is related to MIME types or any other problem.

Environment-Specific Config: Ensure that your application's environment variables and configurations (for development, staging, and production) are set up correctly.

Check GitHub Actions: Ensure that your GitHub Actions workflow is correctly triggering the build process and deploying the built files. You can add debugging output to your workflow to inspect the build and deployment process.

By carefully reviewing these aspects and addressing any misconfigurations, you should be able to resolve the issue with the MIME type and the blank page in your React application.

**Comment by realpinman** [comment]
Hi managed to get this fixed with a little help from the following stack post https://stackoverflow.com/questions/77376027/react-vite-with-typescript-on-azure-server-static-web-app-with-github-reposition

I changed my package.json and staticweb.config which resolved my issue
